libstorage.apib

```diff
@@ -468,33 +531,50 @@ Internal server error
 
     + Schema
 
-            { "$ref": "https://raw.githubusercontent.com/emccode/libstorage/master/libstorage.json#/definitions/internalServerError" }
+            { "$ref": "https://raw.githubusercontent.com/codedellemc/libstorage/master/libstorage.json#/definitions/internalServerError" }
 
 ## Get with Attachments [GET /volumes?{attachments}]
 Gets a list of Volume resources and their attached resources for all configured
 services.
 
 + Parameters
 
-    + attachments (required)
-
-        A flag that indicates that volume attachments should be retrieved.
-        If this flag is provided then the `LibStorage-InstanceID` header must
-        be set -- a comma-delimited list of service name/instance ID,
-        colon-delimited pairs.
+    + attachments (string,optional)
+
+        A value that indicates how and if volume attachments should be retrieved.
+        <br/><br/>
+        The `attachments` parameter can be any of the following values:
+        <br/>
+        Value | Description
+        ------|------------
+        0 | No attachment information is requested. This is the default value and the same as omitting this parameter altogether.
+        1 | Attachment information is requested. This value indicates attachment information should be provided for all volumes.
+        2 | _My_ attachment information is requested. This value indicates attachment information should be provided for volumes attached to the instance specified in the [instance ID](#def-instance-id) request header should be returned. If this bit is set then the instance ID header is required.
+        4 | Attempt to map devices provided via the [local devices](#def-local-devices) request header to the appropriate attachment information. If this bit is set then the instance ID and local device headers are required.
```

Comment:
After reading through the entire PR, I have one outstanding question in this area, combining the local devices header and the `GET /volumes?{attachments}` route...

My understanding is that the LocalDevices header is service/driver specific. E.G. the client provides a list of ScaleIO devices to the ScaleIO driver, EBS devices to the EBS driver, etc. But for `GET /volumes`, there isn't a specific service involved. So what does a client do in that case? Provide a full device map of everything attached?

If *that* was/is the case, it seems better to me to have it built into the libstorage client (rather than an executor) to grab a generic version of this, and then have the storage drivers themselves parse out what is relevant to them (e.g. ScaleIO knows to look at `/dev/scini*`. Not sure if this possible or not (There may be drivers that overlap when it comes to device names?). It seems difficult to handle NFS, iSCSI, kernel block devices, etc. in a generic way, but if this call needs all that info...

Hope what I'm asking makes sense.